lib/main.js

```diff
@@ -0,0 +1,147 @@
+const oracledb = require('oracledb')
+const Store = require('jsreport-sql-store')
+
+module.exports = async (reporter, definition) => {
+  if (reporter.options.store.provider !== 'oracle') {
+    definition.options.enabled = false
+    return
+  }
+
+  function transformBindVarsBoolean (bindVars) {
+    for (let index = 0; index < bindVars.length; index++) {
+      const bindVar = bindVars[index]
+
+      if (bindVar === true) {
+        bindVars[index] = 1
+      } else if (bindVar === false) {
+        bindVars[index] = 0
+      }
+    }
+  }
+
+  function transformResultBoolean (opts, res) {
```

Comment:
This isn't working properly for complex types. This code should work for now.

```js

function transformResultBoolean (row, es, model) { 
  const entityTypeName = model.entitySets[es].entityType.replace(model.namespace + '.', '')
  const entityType = model.entityTypes[entityTypeName]

  for (const propertyName in entityType) {
    const propertyType = entityType[propertyName]

    if (propertyType.type === 'Edm.Boolean') {
      if (row[propertyName] === 1) {
        row[propertyName] = true
      } else if (row[propertyName] === 0) {
        row[propertyName] = false
      }
      continue
    } 

    if (propertyType.complexType) {
      for (const complexColumnName in propertyType.complexType) {
        if (row[propertyName + '_' + complexColumnName] != null) {
          if (propertyType.complexType[complexColumnName].type === 'Edm.Boolean') {
            if (row[propertyName + '_' + complexColumnName] === 1) {
              row[propertyName + '_' + complexColumnName] = true
            } else if (row[propertyName + '_' + complexColumnName] === 0) {
              row[propertyName + '_' + complexColumnName] = false
            }           
          } 
        }
      }
    }
  }
}
```

Invoked as this
```js
if (res.rows && opts.entitySet) {
        for (let row of res.rows) {        
          transformResultBoolean(row, opts.entitySet, reporter.documentStore.model)
        }
}    
```

Working with types will need some general polishing on our side, but for now it should be ok....

Reply:
this test fails
```
common store tests
       internal collection
         insert and query:
     TypeError: Cannot read property 'entityType' of undefined
```
     
the following can be undefined
```
model.entitySets[es].entityType
```

is this expected?

Reply:
Hm, I am sorry. Put there check for now. 

```js
function transformResultBoolean (row, es, model) {
  if (!model.entitySets[es]) {
    return
  }
```

There are some special internal entity sets used in [jsreport-docker-workers](https://github.com/jsreport/jsreport-docker-workers) custom extension, but we will deal with it later.